### PR TITLE
python: mark as externally managed

### PR DESCRIPTION
**Pull request by SMillerDev**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

See https://github.com/orgs/Homebrew/discussions/3404 for context

### CODE REVIEW

**Comment by carlocab** [comment]
https://github.com/Homebrew/homebrew-core/labels/autosquash? Odd.

File: Formula/python@3.11.rb [hunk 1]
```diff
@@ -331,6 +331,22 @@ def install
     }.each do |short_name, long_name|
       (libexec/"bin").install_symlink (bin/long_name).realpath => short_name
     end
+
+    # Mark Homebrew python as externally managed: https://peps.python.org/pep-0668/#marking-an-interpreter-as-using-an-external-package-manager
+    (HOMEBREW_PREFIX/"lib/python#{version.major_minor}/EXTERNALLY-MANAGED").write <<~EOS
```
Comment (carlocab) [inline Formula/python@3.11.rb:336]:
```suggestion
    (site_packages/"EXTERNALLY-MANAGED").write <<~EOS
```
Though I think you may actually need `site_packages_cellar` here? Not sure. @Bo98 might know more.

Comment (carlocab) [reply Formula/python@3.11.rb:336]: Also if you're writing directly to `HOMEBREW_PREFIX` then I think this needs to be in `post_install`.

Comment (SMillerDev) [reply Formula/python@3.11.rb:336]: Yeah, wasn't sure about the location here. I just ended up going with the closest equivalent of upstream docs, but maybe shipping it in the Cellar is better.

Comment (Bo98) [reply Formula/python@3.11.rb:336]: `lib_cellar` looks correct here, based on the documentation that it searches `stdlib`.

**Comment by Bo98** [comment]
I'm wondering whether we should hold on this until 3.12. People may have already installed system-level packages and this could break that. This is also acknowledged in the documentation:

> In particular, we strongly recommend that distros with a concept of major versions only add the marker file or change sysconfig schemes in a new major version; otherwise there is a risk that, on an existing system, software installed via a Python-specific package manager now becomes unmanageable (without an override option). **For a rolling-release distro, if possible, only add the marker file or change sysconfig schemes in a new Python minor version.**

**Comment by carlocab** [comment]
> I'm wondering whether we should hold on this until 3.12

Sounds good to me.

**Pull request closed without merging**